third_party/jitterentropy/jitterentropy-base-user.h

```diff
@@ -112,14 +112,28 @@ static inline void jent_get_nstime(uint64_t *out)
 	*out = EAX_EDX_VAL(val, low, high);
 }
 
-#else /* (__x86_64__) || (__i386__) */
+#elif defined(__aarch64__)
+
+static inline void jent_get_nstime(uint64_t *out)
+{
+        uint64_t ctr_val;
+        /*
+         * Use the system counter for aarch64 (64 bit ARM).
+         */
+        __asm__ volatile("mrs %0, cntvct_el0" : "=r" (ctr_val));
+        *out = ctr_val;
+}
+
+#else /* (__x86_64__) || (__i386__) || (__aarch64__) */
 
 static inline void jent_get_nstime(uint64_t *out)
 {
-	/* OSX does not have clock_gettime -- taken from
-	 * http://developer.apple.com/library/mac/qa/qa1398/_index.html */
 # ifdef __MACH__
-	*out = mach_absolute_time();
+    /* Apple suggests using clock_gettime_nsec_np instead of mach_* functions:
+	 * https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time
+	 * https://developer.apple.com/documentation/kernel/1646199-mach_continuous_time
+	 */
+	*out = clock_gettime_nsec_np(CLOCK_MONOTONIC_RAW);
```

Comment:
`man clock_gettime_nsec_np ` suggests this function call can fail:
```
For clock_gettime_nsec_np() a return value of non-0 indicates success.  A 0
     return value indicates an error occurred and an error code is stored in errno.
```
Should this be handled? At least should be documented why it is ignored.

Reply:
Now when I think about it, should we just delete the whole `__MACH__` branch? Every Apple platform is either `aarch64` or `x86`, both covered with the system counters above.

Reply:
Unless, we hit an Apple 32-bit Arm somewhere? Laptop not. But what's the story on IOS for example?

Reply:
Okay, I don't know the distribution history for iOS. If it doesn't impact anything, I don't see why not.

Reply:
I verified offline that we can remove the special case (32-bit Apple platforms requiring FIPS) and removed it.